text/atacseq-workflow.md

```diff
@@ -0,0 +1,168 @@
+
+# Specification
+
+## Dependencies
+
+- Picard Tools
+- Samtools
+- fq
+- BWA
+- bedtools
+- deeptools
+- TweakSam
+
+## Reference Files
+
+The following reference files are used as the basis of the ATAC-Seq Workflow:
+
+- Similarly to all analysis pipelines in St. Jude Cloud, we use the `GRCh38_no_alt` analysis set for our reference genome. You can get a copy of the file [here](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz). Additionally, you can get the file by running the following commands:
+
+  ```bash
+  wget -o "GRCh38_no_alt.fa.gz" ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna.gz
+
+  echo "a08035b6a6e31780e96a34008ff21bd6  GRCh38_no_alt.fa.gz" > GRCh38_no_alt.fa.gz.md5
+  md5sum -c GRCh38_no_alt.fa.gz.md5
+  # > GRCh38_no_alt.fa.gz: OK
+  ```
+
+- ENCODE list of excluded regions:
+
+  ```bash
+  wget -o "GRCh38_no_alt.exclude.bed.gz" https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg38-blacklist.v2.bed.gz
+  ```
+
+
+## Workflow
+
+Here are the resulting steps in the ATAC-Seq Workflow pipeline. There might be slight alterations in the actual implementation, which can be found in [the St. Jude Cloud workflows repository](https://github.com/stjudecloud/workflows/blob/master/workflows/atacseq/atacseq-standard.wdl).
+
+1. Run `picard ValidateSam` on the incoming BAM to ensure that it is well-formed enough to convert back to FastQ.
+
+    ```bash
+    picard ValidateSamFile \
+                        I=$INPUT_BAM \                     # Input BAM.
+                        IGNORE=INVALID_PLATFORM_VALUE \    # Validations to ignore.
+                        IGNORE=MISSING_PLATFORM_VALUE
+    ```
+
+2. Split BAM file into multiple BAMs on the different read groups using `samtools split`. See [the samtools documentation](http://www.htslib.org/doc/samtools.html) for more information.
+
+    ```bash
+    samtools split -u $UNACCOUNTED_BAM_NAME \ # Reads that do not belong to a read group or the read group is unrecognized go here.
+                    -f '%*_%!.%.'              # Format of output BAM file names.
+    ```
+
+   If the BAM has unaccounted reads, those will need to be triaged and this step will need to be rerun.
+
+3. Run Picard `SamToFastq` on each of the BAMs generated in the previous step.
+
+    ```bash
+        picard SamToFastq \
+                INPUT=$INPUT_BAM \
+                FASTQ=$FASTQ_R1 \
+                SECOND_END_FASTQ=$FASTQ_R2 \
+                RE_REVERSE=true \
+                VALIDATION_STRINGENCY=SILENT
+    ```
+
+4. Run `fq lint` on each of the FastQ pairs that was generated in the previous step as a sanity check. You can see the checks that the `fq` tool performs [here](https://github.com/stjude/fqlib/blob/master/README.md#validators).
```

Comment:
We should be exposing the validators which can be disabled in the WDL. Tangential to this RFC, but you point out the validators here and it would be nice if the WDL exposed them.